### PR TITLE
Adding spracing F3 mini

### PR DESCRIPTION
**Pull request by oleost**
Pulled files from betaflight. 

Added timer in target.c 

And some NAV specific found from regular spracing build with inav. 

Removed transponder ( iNav does have it in drivers,  maybe it should be added as well) 

Also removed #define CONFIG_FASTLOOP_PREFERRED_ACC ACC_NONE from target.h. I guess this is betaflight specific? 

It builds and runs on hardware, but I have not testing any esc or servos or simular on it.. 

Need someone to go over it.

### CODE REVIEW

File: src/main/target/SPRACINGF3MINI/target.h [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+//#define SPRACINGF3MINI_MKII_REVA
+
+#define LED0                    PB3
+
+#define BEEPER                  PC15
+#define BEEPER_INVERTED
+
+#define USE_EXTI
+#define MPU_INT_EXTI            PC13
+#define EXTI15_10_CALLBACK_HANDLER_COUNT 2 // MPU_INT, SDCardDetect
```
Comment (digitalentity) [inline src/main/target/SPRACINGF3MINI/target.h]: This is unnecressary with new IO

**Review by digitalentity — COMMENTED** [review]
Looks good so far. @oleost if you can test that it really works, we can merge this as a part of 1.3 release.

**Comment by oleost** [comment]
Will test as soon as I get my new airplane. 

I don't have any pins on this FC, and I aiming for a very clean and lightweight build so I don't want to solder and desolder stuff on the flight controller before it's in the plane. 

But it should not be far away, I ordered the plane before the FC

**Comment by oleost** [comment]
Got to test more, not happy with compas performance.

Looking at MPU9250 datasheet it contains a AK8963 mag. https://www.invensense.com/products/motion-tracking/9-axis/mpu-9250/

Not an AK8975 which was defined. Tried manually selecting it with mag = 6 but that didnt work. So made changes to taget file to use AK8963 instead, but that gave me error at compile

Change I made https://github.com/oleost/inav/commit/ac3ff16a64959742f322fc942c9664d8f53a5fc3

Error message:

`LD inav_SPRACINGF3MINI.elf
/tmp/ccB3evSE.ltrans22.ltrans.o: In function`detectMag':
/home/ole/oleInav/inav/./src/main/sensors/initialisation.c:622: undefined reference to `ak8963Detect'
collect2: error: ld returned 1 exit status
Makefile:663: recipe for target 'obj/main/inav_SPRACINGF3MINI.elf' failed
make: *** [obj/main/inav_SPRACINGF3MINI.elf] Error 1`

https://github.com/oleost/inav/blob/oleost-SPRACINGF3mini/src/main/sensors/initialisation.c#L622

EDIT: NVM, needed to add it in target.mk file. Succesfully compiled now, dont have time to test as Im on my way out the door, will report later.

**Comment by oleost** [comment]
Still some issues with compass enabled. Looking into issue #590

**Comment by oleost** [comment]
@digitalentity Still having issues with magnetometer. I tried the limiting things I can manage without success. With mag enabled its a lot of drifting. 

Still waiting for my wing before I can try ESC and servo outputs. But as soon as I get it I will try that aswell. And since its an fixed wing Im going to fly with mag disabled anyway.

@digitalentity Do you have an spracingf3mini ? Or maybe @hydra could help out?

**Comment by digitalentity** [comment]
I guess #590 is a similar issue.
Unfortunately I don't have a MINI to troubleshoot the problem.

**Comment by oleost** [comment]
@digitalentity Is there anyway I can help troubleshoot for you?

Also wondering why betaflight have ak8975 configured for spracingf3mini ? Is it a reason or only a mistaken since the MPU9250 datasheet says its an ak8963

**Comment by martinbudden** [comment]
@oleost , I don't have a spracingf3mini either, so I can't do any troubleshooting on this target either.

However, if the datasheet says it has a ak8963, I'd try rebuilding with that compass and see if that solves your problems.

**Comment by oleost** [comment]
@martinbudden I already did it. Have some more troubleshooting to do.

**Comment by digitalentity** [comment]
This doesn't affect anything in the code, just adds a new target. I think it's safe to merge into 1.3 and fix the mag issue later. Mag issue is not exclusive to MINI board, EVO suffers from the same problem.

**Pull request merged**